drivers/soundwire/intel_init.c

```diff
@@ -132,6 +133,13 @@ static int sdw_intel_cleanup(struct sdw_intel_ctx *ctx)
 		if (!ldev->link_res.clock_stop_quirks)
 			pm_runtime_put_noidle(ldev->link_res.dev);
 
+		/*
+		 * Disable runtime PM to prevent peripherals from being resume in the
+		 * link device unregistering process.
+		 */
+		list_for_each_entry(slave, &ldev->link_res.cdns->bus.slaves, node)
+			pm_runtime_disable(&slave->dev);
+
```

Comment:
is this really needed? We already disable pm_runtime in codec drivers, e.g.

https://github.com/thesofproject/linux/blob/59c5a8d96bf6daa9f9cba0437d5830c6c104d11d/sound/soc/codecs/rt711-sdca-sdw.c#L360

Reply:
> is this really needed? We already disable pm_runtime in codec drivers, e.g.
> 
> https://github.com/thesofproject/linux/blob/59c5a8d96bf6daa9f9cba0437d5830c6c104d11d/sound/soc/codecs/rt711-sdca-sdw.c#L360

The resume runtime is called before codec drivers being removed. So, disabling it in the .remove function is too late.

Reply:
alternatively you could do something like we do in the .prepare step:

https://github.com/thesofproject/linux/blob/e915e8b9a14fd32332fd00879eb13c689942c72e/drivers/soundwire/intel_auxdevice.c#L517

But we would need a pm_runtime_resume() which would wait for the devices to be fully resumed (I don't recall why we didn't do this for .prepare).